### PR TITLE
Deprecate span map cache flag

### LINKED ISSUE #5550: Deprecate span map cache flag in slasher
# 🚀 Feature Request
deprecate slasher span map cache flag

### Description
as span map cache flag is essential for slasher use it is time to deprecate this flag and use by default

### PR DESCRIPTION
**Pull request by shayzluf**
Resolves #5550

---

# Description
deprecate span map cache flag

### CODE REVIEW

File: slasher/node/node.go [hunk 1]
```diff
@@ -142,7 +143,7 @@ func (s *SlasherNode) startDB(ctx *cli.Context) error {
 	clearDB := ctx.Bool(cmd.ClearDB.Name)
 	forceClearDB := ctx.Bool(cmd.ForceClearDB.Name)
 	dbPath := path.Join(baseDir, slasherDBName)
-	cfg := &kv.Config{SpanCacheEnabled: ctx.Bool(flags.UseSpanCacheFlag.Name)}
+	cfg := &kv.Config{SpanCacheEnabled: true}
```
[the remark this answers is not in the document]
Comment (shayzluf) [reply slasher/node/node.go]: thanks, removing

**Comment by shayzluf** [comment]
> So this flag was never part of the feature config, is there any reason why you need to add to it there? why not add in deprecated flags in the slasher's flag folders instead and use it from there.

thought of adding the canonical method to deprecate flags

File: shared/featureconfig/flags.go [hunk 2]
```diff
@@ -309,9 +309,17 @@ var (
...
+		Name:   "span-map-cache",
+		Usage:  deprecatedUsage,
+		Hidden: true,
+	}
 )
 
-var deprecatedFlags = []cli.Flag{
+// DeprecatedFlags list of old flags that are not being used in codebase
+// anymore.
+var DeprecatedFlags = []cli.Flag{
```
Comment (nisdas) [inline shared/featureconfig/flags.go:322]: lets not export this, can you follow the same pattern for validator/beacon-node

Comment (shayzluf) [reply shared/featureconfig/flags.go:322]: removed

Comment (shayzluf) [reply shared/featureconfig/flags.go]: fixed

**Pull request merged**